src/main/java/com/velox/sloan/cmo/workflows/kapalibrary/IndexBarcodeToSampleAutoAssigner.java

```diff
@@ -143,12 +156,22 @@ private String getIndexTypesToUse(String taskOptionValueForIndexAssignment) {
      * @throws NotFound
      * @throws ServerException
      */
-    private List<DataRecord> getIndexAssignmentConfigsForIndexType(String indexTypes, List<String> recipes) throws IoError, RemoteException, NotFound, ServerException {
+    private List<DataRecord> getIndexAssignmentConfigsForIndexType(String indexTypes, List<String> recipes, List<DataRecord> attachedSamplesList) throws IoError, RemoteException, NotFound, ServerException {
```

Comment:
I find this logic convoluted, I think a clearer order would be
if (indexTypes.toLowerCase().contains("tcrseq-igo"))
else if (isCrisprOrAmpliconSeq)
else everything else